src/ln/channelmanager.rs

```diff
@@ -910,15 +909,17 @@ impl ChannelManager {
 			}
 		}
 
-		let session_priv = secp_call!(SecretKey::from_slice(&self.secp_ctx, &{
+		let session_priv = SecretKey::from_slice(&self.secp_ctx, &{
 			let mut session_key = [0; 32];
 			rng::fill_bytes(&mut session_key);
 			session_key
-		}));
+		}).expect("RNG is bad!");
 
 		let cur_height = self.latest_block_height.load(Ordering::Acquire) as u32 + 1;
 
-		let onion_keys = ChannelManager::construct_onion_keys(&self.secp_ctx, &route, &session_priv)?;
+		//TODO: This should return something other than HandleError, that's really intended for
+		//p2p-returns only.
+		let onion_keys = secp_call!(ChannelManager::construct_onion_keys(&self.secp_ctx, &route, &session_priv), "Pubkey along hop was maliciously selected", msgs::ErrorAction::IgnoreError);
```

Comment:
Why not extend HandleError further than p2p-returns, something like MarkNodeAsBad { node_id }, could be useful to nodes ban ?

Reply:
Maybe, but best to have more Err types so that you don't end up having match arms that don't make sense in context. eg having a user call send_payment then matching the Err type and ending up with a SendErrMessage action (but they don't know where to send it) is nonsense. Maybe if the full set always makes sense in context but I'm skeptical that'll work out.